packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps, DOMRef, DOMRefValue, FocusEvents, FocusableRef, InputDOMProps, ValueBase} from '@react-types/shared';
+import {centerBaseline} from './CenterBaseline';
+import {ContextValue, Radio, RadioGroup, RadioProps, Provider, RadioGroupStateContext, useSlottedContext} from "react-aria-components"
+import {createContext, forwardRef, ReactNode, useContext, useRef, useCallback, useEffect} from 'react';
+import {focusRing, StyleProps, getAllowedOverrides} from './style-utils' with {type: 'macro'};
+import {IconContext} from './Icon';
+import {style} from '../style/spectrum-theme' with {type: 'macro'};
+import {useDOMRef, useFocusableRef} from '@react-spectrum/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+export interface SegmentedControlProps extends ValueBase<string|null, string>, InputDOMProps, FocusEvents, StyleProps, AriaLabelingProps {
+  /**
+   * The content to display in the segmented control.
+   */
+  children?: ReactNode,
+  /**
+   * Whether the segmented control is disabled.
+   */
+  isDisabled?: boolean
+}
+export interface ControlItemProps extends Omit<RadioProps, 'children' | 'className' | 'style' | 'onHoverStart' | 'onHoverEnd' | 'onHoverChange'>, StyleProps {
+  /**
+   * The content to display in the control item.
+   */
+  children?: ReactNode
+}
+
+export const SegmentedControlContext = createContext<ContextValue<SegmentedControlProps, DOMRefValue<HTMLDivElement>>>(null);
+
+const segmentedControl = style<{size: string}>({
+  font: 'control',
+  display: 'flex',
+  backgroundColor: 'gray-100',
+  borderRadius: 'lg',
+  width: 'full'
+}, getAllowedOverrides())
+
+const controlItem = style({
```

Comment:
Should these have `transition: 'default'`?

Reply:
actually, i added it but not sure how i feel about it. you can compare the previous builds with the first build after this comment and see the difference

Reply:
It looks better IMO, I think we should keep it.

Reply:
talked with spectrum, we're going with a sliding animation instead